From 487e7b37faf81c0af8c94b10bb1418da3cf7629d Mon Sep 17 00:00:00 2001
From: zengchao <chao.zeng@siemens.com>
Date: Sat, 11 Jan 2020 14:17:59 +0800
Subject: [PATCH 13/18] bugfix: set gpio direction output show error

        when gpio pin exceed 31,init the output pin value would show error

        cause:set the gpio pin logical value not setting the correctly gpio bank
        fix:when set the logical value to pin should confirm gpio  bank

Signed-off-by: zengchao <chao.zeng@siemens.com>
---
 drivers/gpio/da8xx_gpio.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/gpio/da8xx_gpio.c b/drivers/gpio/da8xx_gpio.c
index 0deb034504..d5ac4ddbce 100644
--- a/drivers/gpio/da8xx_gpio.c
+++ b/drivers/gpio/da8xx_gpio.c
@@ -15,6 +15,8 @@
 
 #include "da8xx_gpio.h"
 
+static int _gpio_set_value(struct davinci_gpio *bank, unsigned int gpio, int value);
+
 #ifndef CONFIG_DM_GPIO
 #include <asm/arch/hardware.h>
 #include <asm/arch/davinci_misc.h>
@@ -345,7 +347,7 @@ int gpio_free(unsigned int gpio)
 static int _gpio_direction_output(struct davinci_gpio *bank, unsigned int gpio, int value)
 {
 	clrbits_le32(&bank->dir, 1U << GPIO_BIT(gpio));
-	gpio_set_value(gpio, value);
+	_gpio_set_value(bank, (gpio & 0x1f), value);
 	return 0;
 }
 
-- 
2.17.1

